In [5]:
import pandas as pd
# Read csv file

data =pd.read_csv("./concisedata.csv")

In [27]:
# Change the csv file into Dataframe

df = pd.DataFrame(data)
df

COMMUNITY WALL STREET PREDICTION
0     Defect         NaN     Defect
1        NaN      Defect     Defect
2        NaN      Defect     Defect
3     Defect         NaN     Defect
4        NaN   Cooperate  Cooperate
..       ...         ...        ...
72       NaN      Defect  Cooperate
73       NaN      Defect     Defect
74    Defect         NaN  Cooperate
75    Defect         NaN     Defect
76       NaN      Defect     Defect

[77 rows x 3 columns]

In [6]:
# Show data

data

COMMUNITY WALL STREET PREDICTION
0     Defect         NaN     Defect
1        NaN      Defect     Defect
2        NaN      Defect     Defect
3     Defect         NaN     Defect
4        NaN   Cooperate  Cooperate
..       ...         ...        ...
72       NaN      Defect  Cooperate
73       NaN      Defect     Defect
74    Defect         NaN  Cooperate
75    Defect         NaN     Defect
76       NaN      Defect     Defect

[77 rows x 3 columns]

In [12]:
# Select data from 'COMMUNITY' column only

data['COMMUNITY']

0     Defect
1        NaN
2        NaN
3     Defect
4        NaN
       ...  
72       NaN
73       NaN
74    Defect
75    Defect
76       NaN
Name: COMMUNITY, Length: 77, dtype: object

In [17]:
# Drop Empty space

community_data= data['COMMUNITY'].dropna()
community_data


0        Defect
3        Defect
5        Defect
13       Defect
15       Defect
17    Cooperate
18    Cooperate
21       Defect
22       Defect
23       Defect
24       Defect
25       Defect
27    Cooperate
29       Defect
30       Defect
32       Defect
37    Cooperate
40       Defect
43       Defect
46       Defect
50       Defect
55    Cooperate
56    Cooperate
61       Defect
62    Cooperate
63       Defect
64       Defect
67    Cooperate
68       Defect
69    Cooperate
74       Defect
75       Defect
Name: COMMUNITY, dtype: object

In [20]:
# Change the string variable into Dummy variable

dummy_community_data= pd.get_dummies(data['COMMUNITY'].dropna())
dummy_community_data

Cooperate  Defect
0           0       1
3           0       1
5           0       1
13          0       1
15          0       1
17          1       0
18          1       0
21          0       1
22          0       1
23          0       1
24          0       1
25          0       1
27          1       0
29          0       1
30          0       1
32          0       1
37          1       0
40          0       1
43          0       1
46          0       1
50          0       1
55          1       0
56          1       0
61          0       1
62          1       0
63          0       1
64          0       1
67          1       0
68          0       1
69          1       0
74          0       1
75          0       1

In [19]:
# Collect data from 'WALL STREET' game

wallstreet_data= data['WALL STREET'].dropna()
wallstreet_data

1        Defect
2        Defect
4     Cooperate
6     Cooperate
7        Defect
8     Cooperate
9        Defect
10       Defect
11       Defect
12       Defect
14       Defect
16    Cooperate
19       Defect
20       Defect
26    Cooperate
28       Defect
31       Defect
33       Defect
34    Cooperate
35       Defect
36       Defect
38       Defect
39       Defect
41    Cooperate
42       Defect
44    Cooperate
45    Cooperate
47    Cooperate
48       Defect
49    Cooperate
51       Defect
52    Cooperate
53       Defect
54       Defect
57       Defect
58       Defect
59    Cooperate
60       Defect
65       Defect
66    Cooperate
70       Defect
71    Cooperate
72       Defect
73       Defect
76       Defect
Name: WALL STREET, dtype: object

In [21]:
# Change the string variable into Dummy variable

dummy_wallstreet_data= pd.get_dummies(data['WALL STREET'].dropna())
dummy_wallstreet_data

Cooperate  Defect
1           0       1
2           0       1
4           1       0
6           1       0
7           0       1
8           1       0
9           0       1
10          0       1
11          0       1
12          0       1
14          0       1
16          1       0
19          0       1
20          0       1
26          1       0
28          0       1
31          0       1
33          0       1
34          1       0
35          0       1
36          0       1
38          0       1
39          0       1
41          1       0
42          0       1
44          1       0
45          1       0
47          1       0
48          0       1
49          1       0
51          0       1
52          1       0
53          0       1
54          0       1
57          0       1
58          0       1
59          1       0
60          0       1
65          0       1
66          1       0
70          0       1
71          1       0
72          0       1
73          0       1
76          0       1

In [34]:
# In community game

print(dummy_community_data['Cooperate'].mean()) # This is the mean of 'Cooperate' 
print(dummy_community_data['Defect'].mean())    # This is the mean of 'Defect' 

0.28125
0.71875


In [35]:
# To check mean at Wallstreet game

print(dummy_wallstreet_data['Cooperate'].mean()) # This is the mean of 'Cooperate' 
print(dummy_wallstreet_data['Defect'].mean())    # This is the mean of 'Defect' 

0.3333333333333333
0.6666666666666666


In [36]:
# To check whether the mean difference between Community game and wall street game is significant
# We use t-test
# Here we check the significance of 'Cooperate'  value difference between 'Community game' and 'wall street game' 

import scipy.stats

scipy.stats.ttest_ind(dummy_community_data['Cooperate'], dummy_wallstreet_data['Cooperate'], equal_var=False)

Ttest_indResult(statistic=-0.48417838543600356, pvalue=0.6298011255741538)

Interpretation: P- value is over 0.05 So there is no significant difference

In [37]:
# Here we check the significance of 'Defect'  value difference between 'Community game' and 'wall street game' 


scipy.stats.ttest_ind(dummy_community_data['Defect'], dummy_wallstreet_data['Defect'], equal_var=False)

Ttest_indResult(statistic=0.48417838543600406, pvalue=0.6298011255741535)

In [ ]:
Interpretation: P- value is over 0.05 So there is no significant difference

# From now on we will divide our data based on the personality

In [38]:


data

COMMUNITY WALL STREET PREDICTION
0     Defect         NaN     Defect
1        NaN      Defect     Defect
2        NaN      Defect     Defect
3     Defect         NaN     Defect
4        NaN   Cooperate  Cooperate
..       ...         ...        ...
72       NaN      Defect  Cooperate
73       NaN      Defect     Defect
74    Defect         NaN  Cooperate
75    Defect         NaN     Defect
76       NaN      Defect     Defect

[77 rows x 3 columns]

In [39]:

most_likely_to_cooperate = data['PREDICTION'] == 'Cooperate' # Select people that have tendency to cooperate
most_likely_to_defect    = data['PREDICTION'] == 'Defect'    # Select people that have tendency to defect


# Here we start with 'COMMUNITY' game

In [50]:

is_community_game = data['COMMUNITY'] # Select data that are from 'COMMUNITY' game

# Select data that are from Community game and people who have tendency to cooperate
likely_to_cooperate_community= data[most_likely_to_cooperate & is_community_game] 

In [51]:

# Show data that are from Community game with personality cooperate only

likely_to_cooperate_community

COMMUNITY WALL STREET PREDICTION
17  Cooperate         NaN  Cooperate
18  Cooperate         NaN  Cooperate
27  Cooperate         NaN  Cooperate
29     Defect         NaN  Cooperate
37  Cooperate         NaN  Cooperate
40     Defect         NaN  Cooperate
55  Cooperate         NaN  Cooperate
56  Cooperate         NaN  Cooperate
62  Cooperate         NaN  Cooperate
64     Defect         NaN  Cooperate
67  Cooperate         NaN  Cooperate
74     Defect         NaN  Cooperate

In [61]:

is_community_game = data['COMMUNITY'] # Select data that are from 'COMMUNITY' game

# Select data that are from Community game and people who have tendency to 'defect'
likely_to_defect_community= data[most_likely_to_defect & is_community_game]  
likely_to_defect_community

COMMUNITY WALL STREET PREDICTION
0      Defect         NaN     Defect
3      Defect         NaN     Defect
5      Defect         NaN     Defect
13     Defect         NaN     Defect
15     Defect         NaN     Defect
21     Defect         NaN     Defect
22     Defect         NaN     Defect
23     Defect         NaN     Defect
24     Defect         NaN     Defect
25     Defect         NaN     Defect
30     Defect         NaN     Defect
32     Defect         NaN     Defect
43     Defect         NaN     Defect
46     Defect         NaN     Defect
50     Defect         NaN     Defect
61     Defect         NaN     Defect
63     Defect         NaN     Defect
68     Defect         NaN     Defect
69  Cooperate         NaN     Defect
75     Defect         NaN     Defect

In [62]:
# Change the string variable into Dummy variable

dummy_likely_to_cooperate_community = pd.get_dummies(likely_to_cooperate_community['COMMUNITY'])
dummy_likely_to_defect_community    = pd.get_dummies(likely_to_defect_community['COMMUNITY']) 




In [63]:
# Show data that changed into dummy variable ( 'Community game','likely to Cooperate')

dummy_likely_to_cooperate_community

Cooperate  Defect
17          1       0
18          1       0
27          1       0
29          0       1
37          1       0
40          0       1
55          1       0
56          1       0
62          1       0
64          0       1
67          1       0
74          0       1

In [65]:
# Show data that changed into dummy variable ( 'Community game','likely to Defect')

dummy_likely_to_defect_community

Cooperate  Defect
0           0       1
3           0       1
5           0       1
13          0       1
15          0       1
21          0       1
22          0       1
23          0       1
24          0       1
25          0       1
30          0       1
32          0       1
43          0       1
46          0       1
50          0       1
61          0       1
63          0       1
68          0       1
69          1       0
75          0       1

In [68]:
# Show the probabilty about what people choose with 'cooperative' personality

print(dummy_likely_to_cooperate_community['Cooperate'].mean()) 
print(dummy_likely_to_cooperate_community['Defect'].mean())



0.6666666666666666
0.3333333333333333


In [69]:
# Show the probabilty about what people choose with 'defective' personality

print(dummy_likely_to_defect_community['Cooperate'].mean())
print(dummy_likely_to_defect_community['Defect'].mean())

0.05
0.95


# Now we check data from 'Wall street' game

In [81]:

is_wall_street_game = data['WALL STREET'] # Select data that are from 'WALL STREET' game

# Select data that are from Community game and people who have tendency to 'defect'

likely_to_cooperate_wall_street = data[most_likely_to_cooperate & is_wall_street_game] 
likely_to_defect_wall_street    = data[most_likely_to_defect & is_wall_street_game] 

In [74]:
# Show data that are from 'WALL STREET' game with personality cooperate only

likely_to_cooperate_wall_street

COMMUNITY WALL STREET PREDICTION
4        NaN   Cooperate  Cooperate
6        NaN   Cooperate  Cooperate
8        NaN   Cooperate  Cooperate
16       NaN   Cooperate  Cooperate
26       NaN   Cooperate  Cooperate
33       NaN      Defect  Cooperate
34       NaN   Cooperate  Cooperate
41       NaN   Cooperate  Cooperate
44       NaN   Cooperate  Cooperate
45       NaN   Cooperate  Cooperate
47       NaN   Cooperate  Cooperate
49       NaN   Cooperate  Cooperate
52       NaN   Cooperate  Cooperate
59       NaN   Cooperate  Cooperate
66       NaN   Cooperate  Cooperate
70       NaN      Defect  Cooperate
71       NaN   Cooperate  Cooperate
72       NaN      Defect  Cooperate

In [82]:
# Show data that are from 'WALL STREET' game with defective personality  only

likely_to_defect_wall_street

COMMUNITY WALL STREET PREDICTION
1        NaN      Defect     Defect
2        NaN      Defect     Defect
7        NaN      Defect     Defect
9        NaN      Defect     Defect
10       NaN      Defect     Defect
11       NaN      Defect     Defect
12       NaN      Defect     Defect
14       NaN      Defect     Defect
19       NaN      Defect     Defect
20       NaN      Defect     Defect
28       NaN      Defect     Defect
31       NaN      Defect     Defect
35       NaN      Defect     Defect
36       NaN      Defect     Defect
38       NaN      Defect     Defect
39       NaN      Defect     Defect
42       NaN      Defect     Defect
48       NaN      Defect     Defect
51       NaN      Defect     Defect
53       NaN      Defect     Defect
54       NaN      Defect     Defect
57       NaN      Defect     Defect
58       NaN      Defect     Defect
60       NaN      Defect     Defect
65       NaN      Defect     Defect
73       NaN      Defect     Defect
76       NaN      Defect     Defect

In [88]:
# Change the string variable into Dummy variable

dummy_likely_to_cooperate_wall_street = pd.get_dummies(likely_to_cooperate_wall_street['WALL STREET'])
dummy_likely_to_defect_wall_street = pd.get_dummies(likely_to_defect_wall_street['WALL STREET']) 



In [89]:
# Show data that changed into dummy variable ( 'WALL STREET','likely to Cooperate')

dummy_likely_to_cooperate_wall_street 

Cooperate  Defect
4           1       0
6           1       0
8           1       0
16          1       0
26          1       0
33          0       1
34          1       0
41          1       0
44          1       0
45          1       0
47          1       0
49          1       0
52          1       0
59          1       0
66          1       0
70          0       1
71          1       0
72          0       1

In [95]:
# Show data that changed into dummy variable ( 'WALL STREET', 'likely to defect')

dummy_likely_to_defect_wall_street

Defect
1        1
2        1
7        1
9        1
10       1
11       1
12       1
14       1
19       1
20       1
28       1
31       1
35       1
36       1
38       1
39       1
42       1
48       1
51       1
53       1
54       1
57       1
58       1
60       1
65       1
73       1
76       1

In [93]:
# Show the probabilty about what people choose with 'cooperative' personality

print(dummy_likely_to_cooperate_wall_street ['Cooperate'].mean()) 
print(dummy_likely_to_cooperate_wall_street ['Defect'].mean())

0.8333333333333334
0.16666666666666666


In [96]:
# Show the probabilty about what people choose with 'defective' personality

# print(dummy_likely_to_defect_wall_street['Cooperate'].mean()) 
print(dummy_likely_to_defect_wall_street['Defect'].mean())

1.0


Summary

In community game

- Cooperative people tend to cooperate with 66.666% probability
- Cooperative people tend to defect with 33.333% probability

- Defective people tend to cooperate with 5% probability
- Defective people tend to defect with 95% probability


In wall street game

- Cooperative people tend to cooperate with 83.33% probability
- Cooperative people tend to defect with 16.66% probability

- Defective people tend to cooperate with 0% probability
- Defective people tend to defect with 100% probability

From now on, we will also consider personality. We will check whether the difference between community game and wall street game is significant. 


Case 1 Cooperative people




In [97]:
# t-test for checking whether the difference between cooperative people in Community game and Wall street game



scipy.stats.ttest_ind(dummy_likely_to_cooperate_community['Cooperate'], dummy_likely_to_cooperate_wall_street ['Cooperate'], equal_var=False)


Ttest_indResult(statistic=-0.9894733910636695, pvalue=0.3344783337961946)

In [98]:
# t-test for checking whether the difference between cooperative people in Community game and Wall street game

scipy.stats.ttest_ind(dummy_likely_to_cooperate_community['Defect'], dummy_likely_to_cooperate_wall_street ['Defect'], equal_var=False)


Ttest_indResult(statistic=0.9894733910636689, pvalue=0.33447833379619507)

Case 2 Defective people

In [99]:
# t-test for checking whether the difference between defective people in Community game and Wall street game

scipy.stats.ttest_ind(dummy_likely_to_defect_community['Cooperate'], dummy_likely_to_defect_wall_street['Defect'], equal_var=False)

Ttest_indResult(statistic=-19.0, pvalue=8.07462417263228e-14)

Interpretation

We have further divided people based on their personality. After the division, we have checked whether labeling affect people's choice. 

The p-value from t-tests were over 0.05, which means the labeling had no significance